src/main/java/com/dnd/runus/domain/login/service/LoginService.java

```diff
@@ -0,0 +1,84 @@
+package com.dnd.runus.domain.login.service;
+
+import com.dnd.runus.auth.oidc.provider.OidcProviderFactory;
+import com.dnd.runus.auth.token.TokenProviderModule;
+import com.dnd.runus.auth.token.dto.AuthTokenDto;
+import com.dnd.runus.domain.login.dto.request.LoginRequest;
+import com.dnd.runus.domain.login.dto.response.TokenResponse;
+import com.dnd.runus.domain.member.entity.Member;
+import com.dnd.runus.domain.member.entity.PersonalProfile;
+import com.dnd.runus.domain.member.entity.SocialProfile;
+import com.dnd.runus.domain.member.repository.MemberRepository;
+import com.dnd.runus.domain.member.repository.SocialProfileRepository;
+import com.dnd.runus.global.constant.MemberRole;
+import com.dnd.runus.global.constant.SocialType;
+import com.dnd.runus.global.exception.BusinessException;
+import com.dnd.runus.global.exception.type.ErrorType;
+import io.jsonwebtoken.Claims;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@RequiredArgsConstructor
+public class LoginService {
+
+    private final OidcProviderFactory oidcProviderFactory;
+    private final TokenProviderModule tokenProviderModule;
+
+    private final MemberRepository memberRepository;
+    private final SocialProfileRepository socialProfileRepository;
+
+    /**
+     * 회원가입 유뮤 확인 후 회원가입/로그인 진행
+     *
+     * @param request 로그인 request
+     * @return TokenResponse
+     */
+    @Transactional
+    public TokenResponse login(LoginRequest request) {
+
+        Claims claim = oidcProviderFactory.getClaims(request.socialType(), request.idToken());
+        String oAuthId = claim.getSubject();
+        String email = String.valueOf(claim.get("email"));
+
+        // 회원 가입 안되있으면 회원가입 진행
+        if (!isLogin(request.socialType(), oAuthId)) {
+            crateMember(oAuthId, email, request.socialType(), request.nickName());
+        }
+
+        SocialProfile socialProfile = socialProfileRepository
+                .findBySocialTypeAndOauthId(request.socialType(), oAuthId)
+                .orElseThrow(IllegalStateException::new);
```

Comment:
- void에서 private SocialProfile createMember로 바꾸면 SocialProfile findBy 쿼리 두번 안해도 돼요

```suggestion
        SocialProfile socialProfile = socialProfileRepository
                .findBySocialTypeAndOauthId(request.socialType(), oAuthId)
                .orElseGet(() -> createMember(oAuthId, email, request.socialType(), request.nickName()));
```